src/provider/rpc.ts

```diff
@@ -149,48 +158,50 @@ export class RpcProvider implements ProviderInterface {
   ): Promise<BigNumberish> {
     const parsedKey = toHex(toBN(key));
     const block = new Block(blockIdentifier);
-    return this.fetchEndpoint('starknet_getStorageAt', [
-      contractAddress,
-      parsedKey,
-      block.identifier,
-    ]);
+    return this.fetchEndpoint('starknet_getStorageAt', {
+      contract_address: contractAddress,
+      key: parsedKey,
+      block_id: block.identifier,
+    });
   }
 
   // Methods from Interface
-  public async getTransaction(txHash: BigNumberish): Promise<GetTransactionResponse> {
+  public async getTransaction(txHash: string): Promise<GetTransactionResponse> {
     return this.getTransactionByHash(txHash).then(this.responseParser.parseGetTransactionResponse);
   }
 
-  public async getTransactionByHash(
-    txHash: BigNumberish
-  ): Promise<RPC.GetTransactionByHashResponse> {
-    return this.fetchEndpoint('starknet_getTransactionByHash', [txHash]);
+  public async getTransactionByHash(txHash: string): Promise<RPC.GetTransactionByHashResponse> {
+    const transaction_hash = txHash;
+    return this.fetchEndpoint('starknet_getTransactionByHash', { transaction_hash });
```

Comment:
```suggestion
    return this.fetchEndpoint('starknet_getTransactionByHash', { transaction_hash: txHash });
```

And remove `const transaction_hash = txHash;` Is this possible?

Reply:
https://github.com/0xs34n/starknet.js/pull/332/commits/24c24e8060b76f6a300b21c3148a15183b802fcb